File: Deberes_EurekaBank_Arquitectura-entregables/EUREKABANK_RESTFULL_JAVA_BDD_GR06/02 CLIESC/src/main/java/ec/edu/monster/client/model/Movimiento.java
package ec.edu.monster.client.model;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Modelo de datos para movimientos bancarios de EUREKABANK
 * Compatible con todos los servidores (REST Java, REST .NET, SOAP Java, SOAP .NET)
 */
public class Movimiento {
    @JsonProperty("cuenta")
    private String cuenta;

    @JsonProperty("nromov")
    @JsonAlias("nroMov") // Para compatibilidad con .NET
    private int nromov;

    @JsonProperty("fecha")
    private String fecha;

    @JsonProperty("tipo")
    private String tipo;

    @JsonProperty("accion")
    private String accion;

    @JsonProperty("importe")
    private double importe;

    // Constructor vacío para Jackson
    public Movimiento() {}

    // Constructor completo
    public Movimiento(String cuenta, int nromov, String fecha, String tipo, String accion, double importe) {
        this.cuenta = cuenta;
        this.nromov = nromov;
        this.fecha = fecha;
        this.tipo = tipo;
        this.accion = accion;
        this.importe = importe;
    }

    // Getters y Setters
    public String getCuenta() { return cuenta; }
    public void setCuenta(String cuenta) { this.cuenta = cuenta; }

    public int getNromov() { return nromov; }
    public void setNromov(int nromov) { this.nromov = nromov; }

    public String getFecha() { return fecha; }
    public void setFecha(String fecha) { this.fecha = fecha; }

    public String getTipo() { return tipo; }
    public void setTipo(String tipo) { this.tipo = tipo; }

    public String getAccion() { return accion; }
    public void setAccion(String accion) { this.accion = accion; }

    public double getImporte() { return importe; }
    public void setImporte(double importe) { this.importe = importe; }

    @Override
    public String toString() {
        return String.format("Movimiento{cuenta='%s', nromov=%d, fecha='%s', tipo='%s', accion='%s', importe=%.2f}",
                            cuenta, nromov, fecha, tipo, accion, importe);
    }
}

File: EUREKABANK_SOAP_DOTNET_BDD/01 SERVIDOR/Eurekabank_Soap_Dotnet/Eurekabank_Soap_Dotnet/ec.edu.monster.modelo/Movimiento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace Eurekabank_Soap_Dotnet.ec.edu.monster.modelo
{
    [DataContract(Name="movimiento")]
    public class Movimiento
    {
        [DataMember(Order = 1)]
        public string Cuenta { get; set; }

        [DataMember(Order = 2)]
        public int NroMov { get; set; }

        [DataMember(Order = 3)]
        public DateTime Fecha { get; set; }

        [DataMember(Order = 4)]
        public string Tipo { get; set; }

        [DataMember(Order = 5)]
        public string Accion { get; set; }

        [DataMember(Order = 6)]
        public double Importe { get; set; }

        public Movimiento() { }

        public Movimiento(string cuenta, int nromov, DateTime fecha, string tipo, string accion, double importe)
        {
            Cuenta = cuenta;
            NroMov = nromov;
            Fecha = fecha;
            Tipo = tipo;
            Accion = accion;
            Importe = importe;
        }
    }
}